app/uk/gov/hmrc/gform/submission/SubmissionController.scala

```diff
@@ -73,7 +73,12 @@ class SubmissionController(controllerComponents: ControllerComponents, submissio
 
   private def submissionDetailsFormIdData(formIdData: FormIdData, envelopeId: EnvelopeId): Action[AnyContent] =
     formAction("submissionDetailsFormIdData", formIdData) { _ =>
-      submissionService.submissionDetails(SubmissionId(formIdData.toFormId, envelopeId)).asOkJson
+      submissionService
+        .submissionDetails(SubmissionId(formIdData.toFormId, envelopeId))
+        .map {
+          case Some(details) => Ok(Json.toJson(details))
+          case None          => NotFound
```

Comment:
isn't this affecting frontend logic?

Should the GformConnector on frontend [reflect](https://github.com/hmrc/gform-frontend/blob/cm%2FGFORMS-3341/app/uk/gov/hmrc/gform/gformbackend/GformConnector.scala#L348-L371) this change?

This [signature](https://github.com/hmrc/gform-frontend/blob/cm%2FGFORMS-3341/app/uk/gov/hmrc/gform/gformbackend/GformConnector.scala#L348-L351) should change shouldn't it?

```
def submissionDetails(
    formIdData: FormIdData,
    envelopeId: EnvelopeId
  )(implicit ...): Future[Submission] = {
```

Reply:
Correct. I've changed the GformConnector in frontend.